### DISCUSSION TITLE
Enforce quotes on all string values examples in resulting YAML file

### DISCUSSION THREAD

**Discussion by rruimor** [microsoft/typespec#4461] [OpenAPI3]
Hi everyone,

I have a request to have all strings fields examples generated in the OpenAPI YAML file to be enclosed by double quotes. For example if I define:

```typespec
@format("date")
@doc("Date in the format 'YYYY-MM-DD'")
@example("2024-08-26")
scalar Date extends string;
```

It generates the following YAML output:

```yaml
openapi: 3.0.0
info:
  title: (title)
  version: 0.0.0
tags: []
paths: {}
components:
  schemas:
    Date:
      type: string
      example: 2024-08-26
      format: date
      description: Date in the format 'YYYY-MM-DD'
```

The requested scenario would be to generate this:

```yaml
openapi: "3.0.0"
info:
  title: "(title)"
  version: "0.0.0"
tags: []
paths: {}
components:
  schemas:
    Date:
      type: "string"
      example: "2024-08-26"
      format: "date"
      description: "Date in the format 'YYYY-MM-DD'"
```
Where Date example would be "2024-08-26".

You can find the [TypeSpec playground](https://typespec.io/playground?c=QGZvcm1hdCgiZGF0ZSIpCkBkb2MoIkRhdGUgaW4gdGhlIMYhICdZWVlZLU1NLUREJ8QoZXhhbXBsZSgiMjAyNC0wOC0yNiIpCnNjYWxhciDFQGV4dGVuZHMgc3RyaW5nOwoK32zfbMRsJ8ptxBnHbkFub3RoZXLUdQ%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%22options%22%3A%7B%22%40typespec%2Fopenapi3%22%3A%7B%22file-type%22%3A%22yaml%22%7D%7D%7D)

I don't think it's possible at the moment. My current solution is to post-process the yaml file to replace the `example` of type `string`.

Would it make sense to add an emitter option to enforce quotes on string values?

Thanks!

**Comment by timotheeguerin** [comment]
if there was this option wouldn't you expect it to look like this 
```
openapi: "3.0.0"
info:
  title: "(title)"
  version: "0.0.0"
tags: []
paths: {}
components:
  schemas:
    Date:
      type: "string"
      example: "2024-08-26"
      format: "date"
      description: "Date in the format 'YYYY-MM-DD'"
```

Also what is the benefit of doing that, if either way is a valid YAML document? 

Would have to see if the yaml serializer even support that
> Seems like it does 
> defaultStringType | 'BLOCK_FOLDED' ⎮ 'BLOCK_LITERAL' ⎮ 'QUOTE_DOUBLE' ⎮ 'QUOTE_SINGLE' ⎮ 'PLAIN'
-- | --

**Reply by rruimor** [reply]
Thank you for your reply @timotheeguerin. You're right, sorry, it should be all string fields surrounded by double-quotes, let me fix it on my original message.

> Also what is the benefit of doing that, if either way is a valid YAML document?

I wonder the same myself, probably not much. As it is, the output it's valid YAML, but this is a requirement for approving APIs in our team. We started recently using TypeSpec instead of other tools, and this question came.

I'll explain that this is valid YAML as it is, but I wanted to see if this is possible, and if someone would also be interested on this emitter option.

Thanks!
